### PR TITLE
Added installation steps in docs

### PR DESCRIPTION
**Pull request by hdkshingala**
Fixes https://github.com/fluxninja/cloud/issues/5879
Ref https://github.com/fluxninja/cloud/issues/5880

This change is [Reviewable](https://reviewable.io/reviews/fluxninja/aperture/237)

### CODE REVIEW

File: docs/content/setup/installation/agent/daemonset.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+The Aperture Agent can be installed as a
+[Kubernetes DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/),
+where it will get deployed on all the nodes of the cluster.
+
+## Installation {#agent-daemonset-installation}
+
+1. The Aperture Agent can be installed in the DaemonSet mode using the Helm chart of Aperture Operator
+   by using the default `values.yaml` or create a `values.yaml` with below parameters and pass it with `helm upgrade`:
+
+   ```bash
```
Comment (sbienkow-ninja) [inline docs/content/setup/installation/agent/daemonset.md:20]:
```suggestion
   ```yaml
```

File: docs/content/setup/installation/agent/sidecar.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+- If neither label nor annotation is set, the pod is injected if the namespace is listed under
+  `.spec.sidecar.enableNamespacesByDefault`. This is not enabled by default, so generally this
+  means the pod is not injected.
+
+## Installation {#agent-sidecar-installation}
+
+1. The Aperture Agent can be installed in the Sidecar mode using the Helm chart of Aperture Operator
+   by using the default `values.yaml` or create a `values.yaml` with below parameters and pass it with `helm upgrade`:
+
+   ```bash
```
Comment (sbienkow-ninja) [inline docs/content/setup/installation/agent/sidecar.md:36]:
```suggestion
   ```yaml
```

File: docs/content/setup/installation/controller.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+## Installation {#controller-installation}
+
+(Consult [Supported Platforms](setup/supported-platforms.md) before installing.)
+
+**Note**: Make sure you have the [Aperture Operator](setup/installation.md#installation-operator-installation) in running state before following the below steps.
+
+1. The Aperture Controller can be installed using the Helm chart of Aperture Operator
+   by using the default `values.yaml` or create a `values.yaml` with below parameters and pass it with `helm upgrade`:
+
+   ```bash
```
Comment (sbienkow-ninja) [inline docs/content/setup/installation/controller.md:45]:
```suggestion
   ```yaml
```

File: docs/content/setup/installation/installation.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+2. Install or upgrade the chart:
+
+   ```bash
+   helm upgrade --install aperture aperture/operator
+   ```
+
+3. If you want to install just the operator and not the Agent and Controller Custom Resources,
+   create a `values.yaml` with below parameters and pass it with `helm upgrade`:
+
+```bash
```
Comment (sbienkow-ninja) [inline docs/content/setup/installation/installation.md:53]:
```suggestion
```yaml
```

File: docs/content/setup/installation/controller.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+3. Once you have successfully deployed the Custom Resource, confirm that the
+   Aperture Controller is up and running:
+
+```bash
+kubectl get pod -A
+
+kubectl get controller -A
+```
+
+You should see pods for Aperture Controller in `RUNNING` state and `Controller` Custom Resource in `created` state..
```
Comment (sbienkow-ninja) [inline docs/content/setup/installation/controller.md:89]:
```suggestion
You should see pods for Aperture Controller in `RUNNING` state and `Controller` Custom Resource in `created` state.
```

**Review by sbienkow-ninja — APPROVED** [review]
I read through the docs - and apart from few minor issues, all looks great!

File: docs/content/setup/installation/agent/daemonset.md (hunk 1 above)
Comment (hdkshingala) [reply docs/content/setup/installation/agent/daemonset.md]: Fixed.

File: docs/content/setup/installation/agent/sidecar.md (hunk 2 above)
Comment (hdkshingala) [reply docs/content/setup/installation/agent/sidecar.md]: Fixed.

File: docs/content/setup/installation/controller.md (hunk 3 above)
Comment (hdkshingala) [reply docs/content/setup/installation/controller.md]: Fixed.

File: docs/content/setup/installation/installation.md (hunk 4 above)
Comment (hdkshingala) [reply docs/content/setup/installation/installation.md]: Fixed.

File: docs/content/setup/installation/controller.md (hunk 5 above)
Comment (hdkshingala) [reply docs/content/setup/installation/controller.md]: Fixed. Thank you!

**Review by sbienkow-ninja — APPROVED** [review]

**Pull request merged**